### PR TITLE
Checked for required channels in automove.py

### LINKED ISSUE #36: STORY - Housekeeping for automove.py
To ensure the bot does not crash, must modify automove.py to check the constants file to see if the required channels do not exist. If the required channels do not exist in the user's server, auto turn off auto move by writing into the constantvariables.py file and turning off the automove flag. Also print to console that automove cannot be run.

### PR DESCRIPTION
**Pull request by stephenlee91**
- Created a new helper function to check if the required channels in constantvariables.py exist
- If they do not exist when automove is called, open constantvariables.py and write into the file to set AUTO_MOVE_FLAG to False
- Print to console that required channels do not exist and that AUTO_MOVE_FLAG has been set to False
- Return to exit automove function without running the rest of it

### CODE REVIEW

File: cogs/moveusers.py [hunk 1]
```diff
@@ -25,6 +25,15 @@ def _find_similar_channel(channel_name, voice_channels, threshold=0.5):
         return None
 
 
+    # Helper function to check if all required channels for automove are fulfilled
+    def check_required_channels(self, ctx):
```
Comment (TimothyYLin) [inline cogs/moveusers.py]: NIT: Let's prefix the function with `_` to mark "private"

File: cogs/moveusers.py [hunk 2]
```diff
@@ -33,6 +45,7 @@ async def _move_users_to_channel_by_id(self, users, channel_id):
             for user in users:
                 await user.move_to(destination_channel)
 
+
```
Comment (TimothyYLin) [inline cogs/moveusers.py]: NIT: Why newline?

File: cogs/moveusers.py [hunk 3]
```diff
@@ -42,6 +55,16 @@ async def move(self, ctx, member: discord.Member, channel: str):
     # Command function that auto moves users to various different channels depending on activities of all users in all voice channels
     @commands.command()
     async def automove(self, ctx):
+        if not self.check_required_channels(ctx):
+            with open("constantvariables.py", "r+") as file:
+                content = file.read()
+                updated_content = content.replace("AUTO_MOVE_FLAG = True", "AUTO_MOVE_FLAG = False")
+                file.seek(0)
+                file.write(updated_content)
+                file.truncate()
```
Comment (TimothyYLin) [inline cogs/moveusers.py]: Need to close the file.

File: cogs/moveusers.py [hunk 4]
```diff
@@ -42,6 +55,16 @@ async def move(self, ctx, member: discord.Member, channel: str):
     # Command function that auto moves users to various different channels depending on activities of all users in all voice channels
     @commands.command()
     async def automove(self, ctx):
+        if not self.check_required_channels(ctx):
+            with open("constantvariables.py", "r+") as file:
```
Comment (TimothyYLin) [inline cogs/moveusers.py]: Should generally put opening files in try/except in case of FileNotExistError

**Review by TimothyYLin — CHANGES_REQUESTED** [review]

File: cogs/moveusers.py [hunk 5]
```diff
@@ -42,6 +54,19 @@ async def move(self, ctx, member: discord.Member, channel: str):
...
+            try:
+                with open("constantvariables.py", "r+") as file:
+                    content = file.read()
+                    updated_content = content.replace("AUTO_MOVE_FLAG = True", "AUTO_MOVE_FLAG = False")
+                    file.seek(0)
+                    file.write(updated_content)
+                    file.truncate()
+                print("Required channels for using auto move error. Set AUTO_MOVE_FLAG in constantvariables.py to False")
+                return
+            except Exception as e:
```
Comment (LinsaneFC) [inline cogs/moveusers.py:67]: Catch alls are not good

**Review by TimothyYLin — APPROVED** [review]

**Review by asturrus — APPROVED** [review]

**Pull request merged**